### PR TITLE
Add tests to verify edge case and order of colors returned

### LINKED ISSUE #2: Unclear return order for colors in segmentation function
It isn't clear what order the colors are supposed to be returned in for the segmentation function, and we don't have any tests to verify that the order is what we expect.

### PR DESCRIPTION
**Pull request by EricPedley**
# Summary

Added two more unit tests to verify that the segmentation function works with an all-black image, and that the order of colors returned is as we expect it. I also added more description to the docstring of the segmentation function so that it's clear what order the colors are supposed to be returned in.

This PR will close issue #2

### CODE REVIEW

File: src/segmentation.py [hunk 1]
```diff
@@ -5,7 +5,9 @@ def segment_image(img: np.ndarray):
     """
     Get the colors from an image using KMeans clustering
     :param img: The image to get the colors from
-    :return: (mask, colors) where mask  
+    :return: (mask, colors) where mask is a 2D array of the same shape as img, and colors is a 3x3 array of the colors
+    mask is 0 for the background, 1 where the shape is, and 2 where the letter is
```
Comment (Vince-C156) [inline src/segmentation.py:9]: Good change, was confused in the original version what the function was returning.

File: src/segmentation.py [hunk 2]
```diff
@@ -5,7 +5,9 @@ def segment_image(img: np.ndarray):
     """
     Get the colors from an image using KMeans clustering
     :param img: The image to get the colors from
```
Comment (Vince-C156) [inline src/segmentation.py:7]: Maybe be more specific about the input aswell. Those who are not familiar with the test may not know whats coming in as input. Also Specify the input comining in (already segemented from model or just cropped) may also be a good idea.

Comment (Vince-C156) [inline src/segmentation.py]: Maybe be more specific about the input aswell. Those who are not familiar with the test may not know whats coming in as input. Also Specify the input comining in (already segemented from model or just cropped) may also be a good idea.

It may be helpful to also specifiy the color space thats coming in and out of this function aswell as the shape of the image. If it was parsed with opencv it may be bgr. The input image may not always be the same shape, its important to specify however if you know that it will be square or some maximum length and width.

**Review by Vince-C156 — CHANGES_REQUESTED** [review]
I checked out this branch and ran the code. Currently the minimum IOU does not pass the threshold in all cases. Not sure if this was intentional. 

Just make the nessecary changes to the docstring to further detail the funciton.
